cpp/src/arrow/python/datetime.h

```diff
@@ -147,14 +151,25 @@ inline int64_t PyDelta_to_ms(PyDateTime_Delta* pytimedelta) {
 }
 
 ARROW_PYTHON_EXPORT
-inline int64_t PyDelta_to_us(PyDateTime_Delta* pytimedelta) {
-  return (PyDelta_to_s(pytimedelta) * 1000000LL +
-          PyDateTime_DELTA_GET_MICROSECONDS(pytimedelta));
+// TODO which PyDeltas do we cover?
```

Comment:
What's the question exactly?

Reply:
I'd assume the min/max value of `datetime.timedelta` for us and ns. For us its `datetime.timedelta(days=-106751992, seconds=71945, microseconds=224192)` to `datetime.timedelta(days=106751991, seconds=14454, microseconds=775807)` and if i'm not miscalculating `datetime.timedelta(days=-106752, seconds=763, microseconds=145224)` to `datetime.timedelta(days=106751, seconds=85636, microseconds=854775)` for ns

Reply:
Yup! I have also confirmed that at int64, seconds and milliseconds fit without overflow.